transaction/std/src/memo_builder/gift_code_funding_memo_builder.rs

```diff
@@ -297,13 +338,48 @@ mod tests {
     }
 
     #[test]
-    fn test_gift_code_sender_note_builder_creation_fails_with_invalid_note() {
-        // Create Memo Builder with Bad Inputs
+    fn test_gift_code_funding_memo_builder_creation_fails_with_invalid_note() {
+        // Create Memo Builder with note length exceeding max length
         let note_bytes = [b'6'; GiftCodeFundingMemo::NOTE_DATA_LEN + 1];
         let note = std::str::from_utf8(&note_bytes).unwrap();
         let builder = GiftCodeFundingMemoBuilder::new(note);
 
         //Assert memo creation fails
         assert!(matches!(builder, Err(NewMemoError::BadInputs(_))));
     }
+
+    #[test]
+    fn test_gift_code_funding_memo_builder_fee_token_cannot_be_different_from_change_token() {
+        let mut rng: StdRng = SeedableRng::from_seed([0u8; 32]);
+        let gift_code_public_key = RistrettoPublic::from_random(&mut rng);
+        let note = "It's MEMO TIME!!";
+        let mut builder = GiftCodeFundingMemoBuilder::new(note).unwrap();
+        let fee = Amount::new(1, 9001.into());
+
+        // Set a fee with a different token id
+        builder.set_fee(fee).unwrap();
+
+        // Attempt to build the memo
+        let memo_payload = build_gift_code_memos(&mut builder, &gift_code_public_key, fee);
+
+        // Ensure memo creation fails
+        assert!(matches!(memo_payload, Err(NewMemoError::MixedTokenIds)))
```

Comment:
Nit: This can use `assert_eq`
```suggestion
        assert_eq!(memo_payload, Err(NewMemoError::MixedTokenIds))
```

Reply:
Done